### PR TITLE
Set random seed for embedding pipeline (default is None)

### PR DESCRIPTION
**Pull request by RolandJAAI**
Training the Tensorflow Embedding currently gives different results after each run, even if nothing else changed. This can lead to very unstable behavior especially with regards to unseen samples, but also for the training data. For example the user input "ich brauche zahnersatz" yields totally different results after two runs with exactly the same parameters and data. This sample is a part of a slightly longer sample which is in the training data:

rasa_core.processor  - Received user message 'ich brauche zahnersatz' with intent '{'name': 'zahnersatz_zuschuss', 'confidence': 0.909192681312561}' 

rasa_core.processor  - Received user message 'ich brauche zahnersatz' with intent '{'name': 'zahnersatz_zuschuss', 'confidence': 0.6224032044410706}'

The same goes for actual training data like "hi": 
rasa_core.processor  - Received user message 'hi' with intent '{'name': 'greet', 'confidence': 0.9807385206222534}'

rasa_core.processor  - Received user message 'hi' with intent '{'name': 'greet', 'confidence': 0.878046989440918}'

Setting reasonable fallback thresholds (like here at 0.9)  is therefore difficult. Each training run of a production system is a risk. 

This PR changes this and allows to set a stable random seed for Tensorflow and Numpy which can be configured in the pipeline settings. Default is None.

**Proposed changes**:
- added config parameter random_seed to classifier
- added random seed initialization to train method of classifier
- updated changelog
- updated docs
- created test for random seed

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by RolandJAAI** [comment]
@Ghostvv I have made the requested change and also fixed the tests. The changelog already has a general statement saying config and params of this component have been reworked, so this should be ok imo.

**Comment by RolandJAAI** [comment]
implements PR #1506

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible!

**Comment by RolandJAAI** [comment]
@Ghostvv thanks, yes I can look into that. Also the Keras Policy could use a configurable random seed imo sinc with the addition of thresholds and fallback actions reproducible confidence scores matter even more...
